### PR TITLE
fix: logic show modal migration

### PR DESCRIPTION
**Pull request by urmauur**
## Describe Your Changes

-

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: web/containers/Providers/ModalMigrations.tsx [hunk 1]
```diff
@@ -89,9 +88,25 @@ const ModalMigrations = () => {
 
     if (skipMigration) {
       setDidShowMigrationWarning(false)
+    } else {
+      if (!threadsMessagesMigrationSuccess || !modelsMigrationSuccess) {
+        setDidShowMigrationWarning(true)
+        setSkipMigration(false)
```
Comment (louis-jan) [inline web/containers/Providers/ModalMigrations.tsx:94]: skipMigration = false is already set to get into this case, right?

Comment (namchuai) [inline web/containers/Providers/ModalMigrations.tsx:94]: this is quite confusing to me. in this stage, `skipMigration` is `false` and then we still setting `false` here?

File: web/containers/Providers/ModalMigrations.tsx [hunk 2]
```diff
@@ -89,9 +88,23 @@ const ModalMigrations = () => {
...
+  }, [
+    skipMigration,
+    setSkipMigration,
+    setDidShowMigrationWarning,
+    threadsMessagesMigrationSuccess,
+    modelsMigrationSuccess,
+  ])
 
   return (
     <>
```
Comment (namchuai) [inline web/containers/Providers/ModalMigrations.tsx:110]: Should we remove this div? I think `<Modal>` is already enough.

File: electron/handlers/native.ts [hunk 3]
```diff
@@ -352,4 +352,30 @@ export function handleAppIPCs() {
       }
     }
   )
+
+  ipcMain.handle(
+    NativeRoute.getAllLocalModels,
+    async (_event): Promise<any> => {
```
Comment (namchuai) [inline electron/handlers/native.ts:358]: I think `Promise<boolean>` is better, right?

File: web/containers/Providers/ModalMigrations.tsx [hunk 4]
```diff
@@ -69,176 +65,195 @@ const ModalMigrations = () => {
...
-      setThreadsAndMessages(result)
+      const resultThreadsAndMessages = await getJanThreadsAndMessages()
+      const resultLocalModels = await getJanLocalModels()
+      if (
+        resultThreadsAndMessages.threads.length > 0 ||
+        resultThreadsAndMessages.messages.length > 0 ||
+        resultLocalModels.hasLocalModels
+      ) {
+        setDidShowMigrationWarning(true)
+      } else if (step !== 2) {
```
Comment (namchuai) [inline web/containers/Providers/ModalMigrations.tsx:85]: if else if and else have same operation why not just put it in else?

**Review by namchuai — APPROVED** [review]

**Pull request merged**